From dcb0850a2edccd1067385c4a50dd2aab62778009 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Budai?= <ondrej@budai.cz>
Date: Fri, 1 Sep 2023 14:40:06 +0200
Subject: [PATCH] stages/org.osbuild.keymap: create xorg.conf.d if it doesn't
 exist

Newer versions of Fedora don't create this directory by default. This
commit modifies the stage so it creates it if it doesn't exist.
---
 stages/org.osbuild.keymap | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/stages/org.osbuild.keymap b/stages/org.osbuild.keymap
index 7878fa08..c3a292f5 100755
--- a/stages/org.osbuild.keymap
+++ b/stages/org.osbuild.keymap
@@ -18,7 +18,6 @@ be configured for the X11 keyboard.
 Valid keymaps are generally found in /lib/kbd/keymaps.
 """
 
-
 import os
 import subprocess
 import sys
@@ -74,6 +73,7 @@ Section "InputClass"
 EndSection
 """
 
+    os.makedirs(f"{tree}/etc/X11/xorg.conf.d", mode=0o755, exist_ok=True)
     with open(f"{tree}/etc/X11/xorg.conf.d/00-keyboard.conf", "w", encoding="utf8") as f:
         f.write(file_content)
 
-- 
2.43.0

